utils.go

```diff
@@ -83,6 +77,13 @@ func CreateAllResourceLabelsFrom(rm resourceMeta) map[string]string {
 			labels[tag] = field.String()
 		}
 	}
+
+	// Most labels are handled by iterating over the fields of resourceMeta.AzureResource.
+	// Their tag values are used as label keys. The only two labels which are created here
+	// are "resource_group" & "resource_name".
```

Comment:
What about moving `sub_resource_name` assignment also here (and adding to the description)